cassandane/Cassandane/Unit/FormatPretty.pm

```diff
@@ -157,16 +157,24 @@ sub print_errors
     $self->_print($msg);
 
     my $i = 0;
+
+    my $annotations;
+
     for my $e (@{$result->errors()}) {
         my ($test, $errors) = split(/\n/, $e->to_string(), 2);
         chomp $errors;
         my $prettytest = _prettytest($test);
         $self->_print("\n") if $i++;
         $self->_print($self->ansi([31], "$i) $prettytest") . "\n$errors\n");
-        $self->_print("\nAnnotations:\n", $e->object->annotations())
-          if $e->object->annotations();
+
+        # These will always be the same since they share the same test object
+        # so we just need one of them...
+        $annotations ||= \$e->object->annotations();
```

Comment:
I feel like there was a reason this was the way it was, but I don't remember it.  It might just be that I saw the same redundancy and decided to ignore it instead of fix it.  This is one of those things where Cassandane deeply inverts the way Test::Unit wants to work, and then you end up with a mess that's hard to reason about consistently, with little bits of monkey patching everywhere...